src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -70,19 +70,15 @@ private static unsafe void ParseHostEntry(Interop.Sys.HostEntry hostEntry, bool
                     // is likely to involve extra allocations, hashing, etc., and so will probably be more expensive than
                     // this one in the typical (short list) case.
 
-                    var nativeAddresses = new Interop.Sys.IPAddress[hostEntry.IPAddressCount];
+                    Interop.Sys.IPAddress* nativeAddresses = stackalloc Interop.Sys.IPAddress[(int)hostEntry.AddressCount];
```

Comment:
Can we use spans instead of adding unsafe code?

Also, how do we know that the AddressCount will be small enough for this to go on the stack?

Reply:
>Can we use spans instead of adding unsafe code?

I agree we shouldn't be needing to add unsafe code anymore now that `Span<T>` and friends are available.

Reply:
This was a mistake, I lost some changes somewhere in between the multiple repo clones I have. Here should have been `Span` and lower `Slice.IndexOf`.

Anyway, I've changed it back to dynamic allocation. According to [DNS RFC](https://tools.ietf.org/html/rfc1035), it may use TCP (ch 4.2.2) to transmit the data, thus effectively bypassing the UDP limit for the datagram size (ch 2.3.4). So I cannot safely say that it will be small enough to fit on the stack.